containers/ecr-viewer/src/app/services/listEcrDataService.ts

```diff
@@ -57,71 +60,106 @@ export interface EcrDisplay {
  * @param filterConditions - The condition(s) to filter on
  * @returns A promise resolving to a list of eCR metadata
  */
-export async function listEcrData(
+export const listEcrData = audit(
+  "ecr",
+  "query",
+  async (
+    {
+      startIndex,
+      itemsPerPage,
+      sortColumn,
+      sortDirection,
+      filterDates,
+      searchTerm,
+      filterConditions,
+    }: {
+      startIndex: number;
+      itemsPerPage: number;
+      sortColumn: string;
+      sortDirection: string;
+      filterDates: DateRangePeriod;
+      searchTerm?: string;
+      filterConditions?: string[];
+    },
+    trx: Transaction<Core>,
+  ): Promise<EcrDisplay[]> => {
+    const user = await getLoggedInUser();
```

Comment:
this can now take the transaction, otherwise mssql sometimes gets sad

```suggestion
    const user = await getLoggedInUser(trx);
```

Reply:
you could also move it inot the query function since it isn't used out here and avoids adding another variable to pass